pkg/segment/structs/evaluationstructs.go

```diff
@@ -683,8 +689,50 @@ func (self *TextExpr) EvaluateText(fieldToValue map[string]utils.CValueEnclosure
 			}
 		}
 		return minString, nil
-	}
 
+	} else if self.Op == "tostring" {
+		valueStr, err := self.Val.EvaluateToString(fieldToValue)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.Evaluate: failed to evaluate value for 'tostring' operation: %v", err)
+		}
+		boolExpr, err := ParseAndEvalBooleanExpression(valueStr)
+		if err == nil {
+			return strconv.FormatBool(boolExpr), nil
+		}
+		if self.Format != nil {
+			formatStr, err := self.Format.Evaluate(fieldToValue)
+			if err != nil {
+				return "", fmt.Errorf("TextExpr.Evaluate: failed to evaluate format for 'tostring' operation: %v", err)
+			}
+			switch formatStr {
+			case "hex":
+				num, convErr := strconv.Atoi(valueStr)
+				if convErr != nil {
+					return "", fmt.Errorf("TextExpr.Evaluate: failed to convert value '%s' to integer for hex formatting: %v", valueStr, convErr)
+				}
+				return fmt.Sprintf("%#x", num), nil
+			case "commas":
+				num, convErr := strconv.ParseFloat(valueStr, 64)
+				if convErr != nil {
+					return "", fmt.Errorf("TextExpr.Evaluate: failed to convert value '%s' to float for comma formatting: %v", valueStr, convErr)
+				}
+				return fmt.Sprintf("%0.2f", num), nil
```

Comment:
Will this add the commas?

Reply:
No this dosent add commas, as per the documentation it will do this 

tostring(<value>,"commas") | Formats <value> with commas. If the number includes decimals, the function rounds to nearest two decimal places.

Reply:
Ok... so it's supposed to add commas but your code doesn't? Splunk has an example:
```
... | eval n=tostring(1==1) + " " + tostring(15, "hex") + " " + tostring(12345.6789, "commas")
```
which should return `True 0xF 12,345.68` (so 12345.6789 became "12,345.68", so it got rounded and added commas)

Reply:
Okay will do that